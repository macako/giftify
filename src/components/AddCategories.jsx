import { useState } from 'react';

const empty = '';

export const AddCategories = ({ onNewCategory, setCategories }) => {
  const [inputValue, setinputValue] = useState(empty);

  const handleInputChange = ({ target }) => setinputValue(target.value);

  const handleSubmit = (event) => {
    event.preventDefault();

    if (inputValue.trim().length < 1) return;

    //setCategories((categories) => [...categories, inputValue]);
    onNewCategory(inputValue.trim());
    setinputValue(empty);
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        type='text'
        placeholder='Search Gifs'
        value={inputValue}
        onChange={handleInputChange}
      />
    </form>
  );
};
