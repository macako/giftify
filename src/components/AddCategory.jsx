import { useState } from 'react';
import PropTypes from 'prop-types';

const empty = '';

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState(empty);

  const onInputChange = ({ target }) => setInputValue(target.value);

  const onSubmit = (event) => {
    event.preventDefault();

    if (inputValue.trim().length < 1) return;

    onNewCategory(inputValue.trim());
    setInputValue(empty);
  };

  return (
    <form onSubmit={onSubmit} aria-label='form'>
      <input
        type='text'
        placeholder='Search Gifs'
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
