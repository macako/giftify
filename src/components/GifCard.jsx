export const GifCard = ({ title, url }) => {
  return (
    <div className='card'>
      <img src={url} alt={title} />
      <p>{title || 'no title'}</p>
    </div>
  );
};
