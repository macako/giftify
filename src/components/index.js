export * from './AddCategories';
export * from './GifCard';
export * from './GifGrid';
