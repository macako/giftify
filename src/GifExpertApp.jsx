import { useState } from 'react';
import { AddCategories, GifGrid } from './components';

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(['One Punch']);

  const onAddCategory = (newCategory) => {
    if (categories.includes(newCategory)) return;

    return setCategories([newCategory, ...categories]);
  };

  return (
    <>
      <h1>Gif Expert App</h1>
      <AddCategories onNewCategory={onAddCategory} />

      {categories.map((category) => (
        <GifGrid category={category} key={category} />
      ))}
    </>
  );
};
