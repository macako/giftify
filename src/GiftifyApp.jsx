import { useState } from 'react';
import { AddCategory, GifGrid } from './components';

export const GiftifyApp = () => {
  const [categories, setCategories] = useState(['naruto']);

  const onAddCategory = (newCategory) => {
    if (categories.includes(newCategory)) return;

    return setCategories([newCategory, ...categories]);
  };

  return (
    <>
      <h1>Giftify</h1>
      <AddCategory onNewCategory={onAddCategory} />

      {categories.map((category) => (
        <GifGrid category={category} key={category} />
      ))}
    </>
  );
};
