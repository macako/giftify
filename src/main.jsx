import React from 'react';
import ReactDOM from 'react-dom/client';
import { GiftifyApp } from './GiftifyApp.jsx';
import './style.css';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <GiftifyApp />
  </React.StrictMode>
);
