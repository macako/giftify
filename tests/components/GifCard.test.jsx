import { render, screen } from '@testing-library/react';
import { GifCard } from '../../src/components';

describe('<GifCard />', () => {
  const title = 'Naruto';
  const url = 'http://localhost/naruto.jpg';

  test('should match with the snapshot', () => {
    const { container } = render(<GifCard title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test('should display the image with the URL and ALT indicated', () => {
    render(<GifCard title={title} url={url} />);

    const { src, alt } = screen.getByRole('img');

    expect(alt).toBe(title);
    expect(src).toBe(url);
  });

  test('should display title', () => {
    render(<GifCard title={title} url={url} />);

    expect(screen.getByText(title)).toBeTruthy();
  });

  test('should display no title', () => {
    render(<GifCard title={''} url={url} />);

    expect(screen.getByText('no title')).toBeTruthy();
  });
});
