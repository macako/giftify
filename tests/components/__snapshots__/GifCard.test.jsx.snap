// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifCard /> should match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Naruto"
      src="http://localhost/naruto.jpg"
    />
    <p>
      Naruto
    </p>
  </div>
</div>
`;
